rust/arrow/src/array/array.rs

```diff
@@ -2398,30 +2311,23 @@ impl<T: ArrowPrimitiveType> Array for DictionaryArray<T> {
 
     /// Returns the total number of bytes of memory occupied by the buffers owned by this [DictionaryArray].
     fn get_buffer_memory_size(&self) -> usize {
-        self.data.get_buffer_memory_size() + self.values().get_buffer_memory_size()
+        // Since both `keys` and `values` derive (are references from) `data`, we only account for `data`.
+        self.data.get_array_memory_size()
```

Comment:
This part doesn't look true. Am I missing something? Since keys array is also a primitive array, and values are string array both of the arrays' methods for buffer_memory_size should be called and summed up respectively.

Same applies for the array memory size too. Since `zerocopy` is not that much of a zero-copy in the physical memory level it would be nice to explicitly add them in array size calculation too.

Reply:
`data` contains the keys in `buffers[0]` and the values as `child_data[0]`. `self.values` is only another wrapper array, pointing to the same `child_data[0]`.

Reply:
I think that @vertexclique 's point is that we should still use `buffer` from `keys` and `values` here, and in the `array_size`, use `size_of` itself. I think it is a small amount of memory difference, but to keep things consistent, I should change it.

Reply:
@vertexclique , I modified this code. Do you think it is correct now?

Reply:
Voila! Yes.